### PR TITLE
exploitdb: init at 2021-05-22

### LINKED ISSUE #124098: no package for exploitdb
**Project description**
Exploitdb makes it easy to query for exploits based on many parameters. 

**Metadata**

* homepage URL:
https://www.exploit-db.com/
* source URL:
* https://github.com/offensive-security/exploitdb
* license: gplv2
* platforms: unix

### PR DESCRIPTION
**Pull request by applePrincess**
###### Motivation for this change
Add exploitdb

Related to #81418
Fixes #124098

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/security/exploitdb/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    mkdir -p $out/bin
+    cp --recursive ./* $out/bin
+    cp ./.searchsploit_rc $out/bin
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/offensive-security/exploitdb";
+    description = "Archive of public exploits and corresponding vulnerable software, developed for use by penetration testers and vulnerability researchers.";
+    license = lib.licenses.gpl2;
+    maintainer = [ lib.maintainers.applePrincess ];
```
Comment (fabaff) [inline pkgs/tools/security/exploitdb/default.nix:22]:
```suggestion
    maintainer = [ maintainers.applePrincess ];
```

File: pkgs/tools/security/exploitdb/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  installPhase = ''
+    mkdir -p $out/bin
+    cp --recursive ./* $out/bin
+    cp ./.searchsploit_rc $out/bin
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/offensive-security/exploitdb";
+    description = "Archive of public exploits and corresponding vulnerable software, developed for use by penetration testers and vulnerability researchers.";
+    license = lib.licenses.gpl2;
```
Comment (fabaff) [inline pkgs/tools/security/exploitdb/default.nix:21]: Headers contains other licenses like [GPL3+](https://github.com/offensive-security/exploitdb/blob/master/exploits/php/webapps/11263.php) or [MIT](https://github.com/offensive-security/exploitdb/blob/master/shellcodes/linux_x86-64/41503.nasm). I assume that there could me more.

```suggestion
    license = with licenses; [ gpl2Plus gpl3Plus mit ];
```

File: pkgs/tools/security/exploitdb/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{stdenv, lib, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  name = "exploitdb";
+  version = "2021-05-22";
+  src = fetchFromGitHub {
```
Comment (fabaff) [inline pkgs/tools/security/exploitdb/default.nix:6]: Better readability for humans :wink: 

```suggestion
  version = "2021-05-22";

  src = fetchFromGitHub {
```

File: pkgs/tools/security/exploitdb/default.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    owner = "offensive-security";
+    repo = name;
+    rev = version;
+    sha256 = "sha256-cpzAdRAtF47j92u8GHyu6V1ycqtNld+9saCqoboAu+o=";
+  };
+  installPhase = ''
```
Comment (fabaff) [inline pkgs/tools/security/exploitdb/default.nix:12]:
```suggestion
  };

  installPhase = ''
```

File: pkgs/tools/security/exploitdb/default.nix [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{stdenv, lib, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  name = "exploitdb";
```
Comment (legendofmiracles) [inline pkgs/tools/security/exploitdb/default.nix:4]:
```suggestion
  pname = "exploitdb";
```

**Comment by legendofmiracles** [comment]
Result of `nixpkgs-review pr 124209` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>exploitdb</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 124209` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>exploitdb</li>
  </ul>
</details>

**Review by fabaff — COMMENTED** [review]
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] package built (see https://github.com/NixOS/nixpkgs/pull/124209#issuecomment-848112525)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
...ew/pr-124209-1]$ ./results/exploitdb/bin/searchsploit -e pi3web
[...]
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- ---------------------------------
 Exploit Title                                                                                                                                                                                                             |  Path
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- ---------------------------------
John Roy Pi3Web 1.0.1 - Buffer Overflow                                                                                                                                                                                    | windows/dos/20634.txt
John Roy Pi3Web 2.0 For Windows - Remote Buffer Overflow                                                                                                                                                                   | windows/remote/21225.c
Pi3Web 2.0.1 - Denial of Service (PoC)                                                                                                                                                                                     | windows/dos/22.c
Pi3Web 2.0.1 - GET Denial of Service                                                                                                                                                                                       | windows/dos/22587.c
Pi3Web 2.0.2 - SortName Buffer Overflow                                                                                                                                                                                    | windows/dos/22718.c
Pi3Web 2.0.3 - 'ISAPI' Remote Denial of Service                                                                                                                                                                            | windows/dos/7109.txt
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- ---------------------------------
Shellcodes: No Results
[...]           
```

File: pkgs/tools/security/exploitdb/default.nix [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    mkdir -p $out/bin
+    cp --recursive ./* $out/bin
+    cp ./.searchsploit_rc $out/bin
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/offensive-security/exploitdb";
+    description = "Archive of public exploits and corresponding vulnerable software";
+    license = with licenses; [gpl2Plus gpl3Plus mit];
+    maintainers = [ maintainers.applePrincess ];
```
Comment (SuperSandro2000) [inline pkgs/tools/security/exploitdb/default.nix:24]:
```suggestion
    license = with licenses; [ gpl2Plus gpl3Plus mit ];
    maintainers = with maintainers; [ applePrincess ];
```

**Pull request merged**